apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/V1/SpannerDateTests.cs

```diff
@@ -0,0 +1,193 @@
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using Xunit;
+
+namespace Google.Cloud.Spanner.V1.Tests
+{
+    public class SpannerDateTests
+    {
+        private static readonly string MaxDate = DateTime.MaxValue.Date.ToString("yyyy-MM-dd", CultureInfo.InvariantCulture);
+
+        private static readonly string MinDate = DateTime.MinValue.Date.ToString("yyyy-MM-dd", CultureInfo.InvariantCulture);
+
+        public static IEnumerable<object[]> FromDateTimeData()
+        {
+            // Input: DateTime values.
+            yield return new object[] { new DateTime(2022, 4, 18, 0, 0, 0, DateTimeKind.Utc) };
+            yield return new object[] { new DateTime(2022, 4, 18, 1, 0, 0, DateTimeKind.Utc) };
+            yield return new object[] { new DateTime(2022, 4, 18, 2, 0, 0, DateTimeKind.Utc) };
+            yield return new object[] { new DateTime(2022, 4, 18, 3, 0, 0, DateTimeKind.Utc) };
+            yield return new object[] { new DateTime(2022, 4, 18, 4, 0, 0, DateTimeKind.Utc) };
+            yield return new object[] { new DateTime(2022, 4, 18, 5, 0, 0, DateTimeKind.Utc) };
+            yield return new object[] { new DateTime(2022, 4, 18, 0, 0, 0, DateTimeKind.Local) };
+            yield return new object[] { new DateTime(2022, 4, 18, 1, 0, 0, DateTimeKind.Local) };
+            yield return new object[] { new DateTime(2022, 4, 18, 2, 0, 0, DateTimeKind.Local) };
+            yield return new object[] { new DateTime(2022, 4, 18, 3, 0, 0, DateTimeKind.Local) };
+            yield return new object[] { new DateTime(2022, 4, 18, 4, 0, 0, DateTimeKind.Local) };
+            yield return new object[] { new DateTime(2022, 4, 18, 5, 0, 0, DateTimeKind.Local) };
+            yield return new object[] { new DateTime(2022, 4, 18, 23, 59, 59, DateTimeKind.Utc) };
+            yield return new object[] { new DateTime(2022, 4, 18, 23, 59, 59, DateTimeKind.Local) };
+            yield return new object[] { new DateTime(2022, 4, 18, 23, 59, 59, DateTimeKind.Unspecified) };
+            yield return new object[] { new DateTime(0001, 01, 01, 0, 0, 0, DateTimeKind.Utc) };
+            yield return new object[] { new DateTime(9999, 12, 31, 23, 59, 59, DateTimeKind.Utc) };
+            yield return new object[] { new DateTime(2000, 12, 31, 23, 59, 59, DateTimeKind.Unspecified) };
+        }
+
+        [Theory]
+        [InlineData("9999-12-31")]
+        [InlineData("0001-01-01")]
+        [InlineData("2022-04-20")]
+        public void Parse_Valid(string input)
+        {
+            var date = SpannerDate.Parse(input);
+            var success = SpannerDate.TryParse(input, out SpannerDate date2);
+            Assert.True(success);
+            Assert.Equal(date, date2);
+        }
+
+        [Theory]
+        [InlineData("0001-01-01")]
+        [InlineData("9999-12-31")]
+        [InlineData("2022-04-18")]
+        [InlineData("2020-02-29")]
+        public void ParseToStringRoundTrip(string input)
```

Comment:
This test and the one above can be put together now - because *everything* should roundtrip.